main.go

```diff
@@ -1,426 +0,0 @@
-		if err != nil {
-			klog.Fatalf("Failed to instantiate delegating authenticator: %v", err)
-		}
-
-		go delegatingAuthenticator.Run(1, context.Background().Done())
-		authenticator = delegatingAuthenticator
-	}
-
-	sarClient := kubeClient.AuthorizationV1().SubjectAccessReviews()
-	sarAuthorizer, err := authz.NewSarAuthorizer(sarClient)
-
-	if err != nil {
-		klog.Fatalf("Failed to create sar authorizer: %v", err)
-	}
-
-	staticAuthorizer, err := authz.NewStaticAuthorizer(cfg.auth.Authorization.Static)
-	if err != nil {
-		klog.Fatalf("Failed to create static authorizer: %v", err)
-	}
-
-	authorizer := union.New(
-		staticAuthorizer,
-		sarAuthorizer,
-	)
-
-	auth, err := proxy.New(kubeClient, cfg.auth, authorizer, authenticator)
-
-	if err != nil {
-		klog.Fatalf("Failed to create rbac-proxy: %v", err)
-	}
-
-	upstreamTransport, err := initTransport(cfg.upstreamCAFile)
-	if err != nil {
-		klog.Fatalf("Failed to set up upstream TLS connection: %v", err)
-	}
-
-	if len(cfg.allowPaths) > 0 && len(cfg.ignorePaths) > 0 {
-		klog.Fatal("Cannot use --allow-paths and --ignore-paths together.")
-	}
-
-	for _, pathAllowed := range cfg.allowPaths {
-		_, err := path.Match(pathAllowed, "")
-		if err != nil {
-			klog.Fatalf("Failed to verify allow path: %s", pathAllowed)
-		}
-	}
-
-	for _, pathIgnored := range cfg.ignorePaths {
-		_, err := path.Match(pathIgnored, "")
-		if err != nil {
-			klog.Fatalf("Failed to verify ignored path: %s", pathIgnored)
-		}
-	}
-
-	proxy := httputil.NewSingleHostReverseProxy(upstreamURL)
-	proxy.Transport = upstreamTransport
-
-	if cfg.upstreamForceH2C {
-		// Force http/2 for connections to the upstream i.e. do not start with HTTP1.1 UPGRADE req to
-		// initialize http/2 session.
-		// See https://github.com/golang/go/issues/14141#issuecomment-219212895 for more context
-		proxy.Transport = &http2.Transport{
-			// Allow http schema. This doesn't automatically disable TLS
-			AllowHTTP: true,
-			// Do disable TLS.
-			// In combination with the schema check above. We could enforce h2c against the upstream server
-			DialTLS: func(netw, addr string, cfg *tls.Config) (net.Conn, error) {
-				return net.Dial(netw, addr)
-			},
-		}
-	}
-
-	mux := http.NewServeMux()
-	mux.Handle("/", http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
-		found := len(cfg.allowPaths) == 0
-		for _, pathAllowed := range cfg.allowPaths {
-			found, err = path.Match(pathAllowed, req.URL.Path)
-			if err != nil {
-				return
-			}
-			if found {
-				break
-			}
-		}
-		if !found {
-			http.NotFound(w, req)
-			return
-		}
-
-		ignorePathFound := false
-		for _, pathIgnored := range cfg.ignorePaths {
-			ignorePathFound, err = path.Match(pathIgnored, req.URL.Path)
-			if err != nil {
-				return
-			}
-			if ignorePathFound {
-				break
-			}
-		}
-
-		if !ignorePathFound {
-			ok := auth.Handle(w, req)
-			if !ok {
-				return
-			}
-		}
-
-		proxy.ServeHTTP(w, req)
```

Comment:
Flag for diff against the kube-apiserver upgradeaware proxy.  I'm surprised a different proxy is being used here than the aggregated apiserver proxy.  I remember issues with ugprade awareness.